### DISCUSSION TITLE
Create db from macro

### DISCUSSION THREAD

**Discussion by Ludea** [diesel-rs/diesel#3433] [Q&A] [answered]
Hi, I read the Quick start guide.
We need to install diesel cli to create database and migration script.

Can we do the same thing, with macros, or fn ?

**Comment by weiznich** [comment, marked as answer]
Executing migrations is supported by the `diesel_migrations` crate. Creating the database is a single SQL command (`CREATE DATABASE your_db_name`), which can be executed by using `diesel::sql_query`.

(`diesel_cli` internally does the same thing)

**Reply by Ludea** [reply]
And what about creating Sqlite file ?

**Reply by weiznich** [reply]
That's automatically handled as soon as you try to open a `SqliteConnection` to an non-existing database.

**Reply by Ludea** [reply]
Perfect thanks for clarification
